### PR TITLE
Add sorted troves

### PR DESCRIPTION
**Pull request by phamminh0811**

### CODE REVIEW

File: contracts/sorted-troves/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+[features]
+# for more explicit tests, cargo test --features=backtraces
+backtraces = ["cosmwasm-std/backtraces"]
+# use library feature to disable all instantiate/execute/query exports
+library = []
+
+[dependencies]
+cosmwasm-schema = "1.1.3"
+cosmwasm-std = {version = "1.1.3" }
```
Comment (expertdicer) [inline contracts/sorted-troves/Cargo.toml]: Use `cosmwasm-sdt`  version 1.0.0 or upgrade all other cargo.toml

File: contracts/sorted-troves/src/contract.rs [hunk 2]
```diff
@@ -0,0 +1,522 @@
...
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    mut deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+    // set admin so that only admin can access to update role function
+    let api = deps.api;
```
Comment (expertdicer) [inline contracts/sorted-troves/src/contract.rs:27]: Redundant

File: contracts/sorted-troves/src/contract.rs [hunk 3]
```diff
@@ -0,0 +1,522 @@
...
+    let mut next_id = next_id;
+    if prev_id.is_some() {
+        let prev_id_nicr: Uint256 = deps.querier.query_wasm_smart(
+            trove_manager.clone(),
+            &ultra_base::trove_manager::QueryMsg::GetNominalICR { 
+                borrower:  prev_id.clone().unwrap().to_string()
+            }
+        )?;
+        if NODES.may_load(deps.storage, prev_id.clone().unwrap())?.is_none()
+            && nicr > prev_id_nicr{
```
Comment (expertdicer) [inline contracts/sorted-troves/src/contract.rs]: Check logic

File: contracts/sorted-troves/src/contract.rs [hunk 4]
```diff
@@ -0,0 +1,522 @@
...
+
+    if next_id.is_some() {
+        let next_id_nicr: Uint256 = deps.querier.query_wasm_smart(
+            trove_manager.clone(),
+            &ultra_base::trove_manager::QueryMsg::GetNominalICR { 
+                borrower:  next_id.clone().unwrap().to_string()
+            }
+        )?;
+        if NODES.may_load(deps.storage, next_id.clone().unwrap())?.is_none()
+            && nicr < next_id_nicr{
```
Comment (expertdicer) [inline contracts/sorted-troves/src/contract.rs]: Recheck Logic

File: contracts/sorted-troves/Cargo.toml (hunk 1 above)
Comment (phamminh0811) [reply contracts/sorted-troves/Cargo.toml]: cosmwasm std version 1.19 still work on juno, so it's still ok

**Comment by expertdicer** [comment]
LGTM!

**Pull request merged**